test/jdk/java/util/UUID/UUIDTest.java

```diff
@@ -22,22 +22,27 @@
  */
 
 /* @test
- * @bug 4173528 5068772 8148936 8196334
+ * @bug 4173528 5068772 8148936 8196334 8308803
  * @summary Unit tests for java.util.UUID
  * @key randomness
- * @run main/othervm -XX:+CompactStrings UUIDTest
- * @run main/othervm -XX:-CompactStrings UUIDTest
+ * @run main/othervm -Xmx1g -XX:+CompactStrings UUIDTest
+ * @run main/othervm -Xmx1g -XX:-CompactStrings UUIDTest
  */
 
 import java.util.*;
+import java.util.stream.IntStream;
 
 public class UUIDTest {
 
-    static Random generator = new Random();
+    // Single UUID instance is ~32 bytes, 1M instances take ~256M in the set
+    private static final int COUNT = 1_000_000;
+
+    static final Random generator = new Random();
```

Comment:
Hello Aleksey, I realize this is unrelated to the changes in this PR, but since we are updating this test, would using `jdk.test.lib.RandomFactory.getRandom()` from test library be a better idea here? That would then print the seed in the logs and could help debug any collisions more easily?

Reply:
Yes, good idea, added.